components/Common/OramaSearchBar/search.ts

```diff
@@ -0,0 +1,26 @@
+import { search } from '@orama/orama';
+
+import type { Document, Orama, RawData, Schema } from '@orama/orama';
+import type { SearchResult } from '../../../types';
+
+export type DocumentMapper = (doc: Document) => SearchResult;
+
+export type OramaSearchBarProps = {
+  schema: Schema;
+  index: RawData;
+  documentMapper?: DocumentMapper;
+};
+
+export const identityDocumentMapper = (doc: Document): SearchResult => {
```

Comment:
The default one is technically a no-op.

Similarly to `SearchBar`, I designed those functions to be use-agnostic. 
Depending on when you're rendering results, you might need to map Orama results to something else. I left room for a simple mapping function to be passed, rather than implementing this inside `useEffect` or similar.